src/metabase/lib/filter/update.cljc

```diff
@@ -217,8 +237,13 @@
     longitude-column                             :- :some
     {:keys [north east south west], :as _bounds} :- [:ref ::lat-lon.bounds]]
    (-> query
-       (remove-existing-filters-against-column stage-number latitude-column)
-       (remove-existing-filters-against-column stage-number longitude-column)
-       (lib.filter/filter stage-number (let [[lat-min lat-max] (sort [north south])
-                                             [lon-min lon-max] (sort [east west])]
-                                         (lib.filter/inside latitude-column longitude-column lat-max lon-min lat-min lon-max))))))
+       (remove-existing-filters-against-column-checking-subclauses stage-number latitude-column)
+       (remove-existing-filters-against-column-checking-subclauses stage-number longitude-column)
+       (lib.filter/filter stage-number
+                          (if (<= west east)
+                            ;; bounds do not cross the antimerdian. A single :inside filter suffices.
+                            (lib.filter/inside latitude-column longitude-column north west south east)
+                            ;; bounds do cross the antimerdian. Split into two filters for the east and west sides.
+                            (lib.filter/or
+                             (lib.filter/inside latitude-column longitude-column north west south 180)
+                             (lib.filter/inside latitude-column longitude-column north -180 south east)))))))
```

Comment:
We could instead "factor out" the common latitude filter like so

```clojure
(lib.filter/and
   (lib.filter/between latitude-column south north)
   (lib.filter/or
    (lib.filter/between longitude-column west   180.0)
    (lib.filter/between longitude-column -180.0 east)))
```

which results in more compact SQL, but also in two filter pills in the UI that the user has to clear separately in order to remove the filter.